src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/SourcelinkTests.cs

```diff
@@ -0,0 +1,134 @@
+    public void VerifySourcelinks()
+    {
+        if (Directory.Exists(SourcelinkRoot))
+        {
+            Directory.Delete(SourcelinkRoot, true);
+        }
+        Directory.CreateDirectory(SourcelinkRoot);
+
+        sourcelinkToolPath = GetSourcelinkToolPath();
+
+        ValidateAllFiles(ExtractPackages(GetAllSymbolsPackages()));
+
+        foreach (string file in failedFiles)
+        {
+            OutputHelper.WriteLine($"Failed sourcelink verification: {file}");
+        }
+
+        Assert.True(failedFiles.Count == 0);
+    }
+
+    /// <summary>
+    /// Gets the path to sourcelink binary.
+    /// Extracts the dotnet-sourcelink tool package from PSB arhive.
+    /// </summary>
+    /// <returns>Path to sourcelink tool binary.</returns>
+    private string GetSourcelinkToolPath()
+    {
+        string sourcelinkToolPackageNamePattern = "dotnet-sourcelink*nupkg";
+        string sourcelinkToolBinaryFilenamePattern = "dotnet-sourcelink.dll";
+
+        string sourcelinkRoot = Directory.CreateDirectory(Path.Combine(SourcelinkRoot, "sourcelink-tool")).FullName;
+        Utilities.ExtractTarball(Config.SourceBuiltArtifactsPath, sourcelinkRoot, sourcelinkToolPackageNamePattern);
+        string[] files = Directory.GetFiles(sourcelinkRoot, sourcelinkToolPackageNamePattern, SearchOption.AllDirectories);
+        Assert.True(files.Length > 0, "Did not find sourcelink tool package in PSB Artifacts archive");
+
+        string extractedToolPath = Directory.CreateDirectory(Path.Combine(sourcelinkRoot, "extracted")).FullName;
+        Utilities.ExtractNupkg(files[0], extractedToolPath);
+
+        files = Directory.GetFiles(extractedToolPath, sourcelinkToolBinaryFilenamePattern, SearchOption.AllDirectories);
+        Assert.True(files.Length > 0, $"Did not find sourcelink tool binary with expected filename pattern: {sourcelinkToolBinaryFilenamePattern}");
+
+        return files[0];
+    }
+
+    private IEnumerable<string> GetAllSymbolsPackages()
+    {
+        /*
+            At the moment we validate sourcelinks from runtime symbols package.
+            The plan is to make symbols, from all repos, available in source-build artifacts.
+            Once that's available, this code will be modified to validate all available symbols.
+            Tracking issue: https://github.com/dotnet/source-build/issues/3612
+        */
+
+        // Runtime symbols package lives in the same directory as PSB artifacts.
+        // i.e. <repo-root>/artifacts/x64/Release/runtime/dotnet-runtime-symbols-fedora.36-x64-8.0.0-preview.7.23355.7.tar.gz
+        string runtimeSymbolsPackageNamePattern = "dotnet-runtime-symbols-*.tar.gz";
+        string[] files = Directory.GetFiles(Path.GetDirectoryName(Config.SourceBuiltArtifactsPath), runtimeSymbolsPackageNamePattern, SearchOption.AllDirectories);
+        Assert.True(files.Length > 0, "Did not find runtime symbols archive");
+
+        yield return files[0];
```

Comment:
Why is the return type `IEnumerable<string>` but only ever returning one item?

Reply:
We'll be updating this code soon, to enumerate all symbols archives.

Reply:
Is this work still pending?

Reply:
It is coming up very soon. PR for creating symbols archives will be out today or tomorrow. Update to sourcelink validation to consume those archives will follow few days later, barring any unforeseen issues.